test/gtest/common/mem_buffer.cc

```diff
@@ -43,20 +43,41 @@
 #endif
 
 
+bool mem_buffer::is_cuda_supported()
+{
+#if HAVE_CUDA
+    int num_gpus;
+    cudaError_t cudaErr = cudaGetDeviceCount(&num_gpus);
+    return (cudaErr == cudaSuccess) || (num_gpus > 0);
+#else
+    return false;
+#endif
+}
+
+bool mem_buffer::is_rocm_supported()
+{
+#if HAVE_ROCM
+    hsa_status_t status = hsa_init();
```

Comment:
@yosefe This breaks build with rocm and gtest enabled because the test suite only links to hip runtime (equivalent to cuda runtime) and not the hsa runtime (equivalent to cuda driver). I will push a PR to fix this. Sorry I did not see this PR earlier.

Reply:
sorry about that, thanks for the catch!